cpp/include/cugraph/algorithms.hpp

```diff
@@ -1752,17 +1760,18 @@ std::tuple<rmm::device_uvector<vertex_t>,
            std::optional<rmm::device_uvector<edge_t>>,
            std::optional<rmm::device_uvector<edge_type_t>>,
            rmm::device_uvector<int32_t>,
-           std::optional<rmm::device_uvector<int32_t>>>
+           std::optional<rmm::device_uvector<int32_t>>,
+           std::optional<rmm::device_vector<edge_t>>>
 uniform_neighbor_sample(
   raft::handle_t const& handle,
   graph_view_t<vertex_t, edge_t, store_transposed, multi_gpu> const& graph_view,
   std::optional<edge_property_view_t<edge_t, weight_t const*>> edge_weight_view,
-  std::optional<
-    edge_property_view_t<edge_t,
-                         thrust::zip_iterator<thrust::tuple<edge_t const*, edge_type_t const*>>>>
-    edge_id_type_view,
+  std::optional<edge_property_view_t<edge_t, edge_t const*>> edge_id_view,
+  std::optional<edge_property_view_t<edge_t, edge_type_t const*>> edge_type_view,
   rmm::device_uvector<vertex_t>&& starting_vertices,
   std::optional<rmm::device_uvector<int32_t>>&& starting_labels,
+  std::optional<rmm::device_uvector<edge_t>>&& starting_vertex_offsets,
+  std::optional<rmm::device_uvector<int32_t>>&& batch_id_to_output_gpu_mapping,
```

Comment:
And can we reconstruct a tree per starting vertex with the current API?

Reply:
We had explicitly tried to avoid "batch ID" in the previous API, largely because the previous implementation had some baggage associated with it.  Since we've deleted that code, I'd be happy to switch back to batch id if that feels more intuitive.

I'm not sure if we should keep the old technique or just migrate to the new technique (passing CSR-like structure).  I'd be OK changing the API to use `starting_vertex_offsets` and eliminating `starting_labels`.  I was thinking it might be helpful to transition (support both, drop the `starting_labels` later).

Reply:
Perhaps the more detailed discussion below should inform this choice.

Reply:
Switching back to "label" in the next push.